test/jwtAuthorization.js

```diff
@@ -0,0 +1,301 @@
+'use strict';
+require('./config');
+const {
+    describe,
+    it, beforeEach, afterEach, before, after
+} = require('mocha');
+const assert = require('assert');
+const Koa = require('koa');
+const koaBody = require('koa-body');
+const routerFactory = require('koa-router');
+const { JWTIssuer } = require('@storyous/storyous-jwt');
+const JWTMockServer = require('./JWTMockServer');
+const { jwtPermissions, fetch } = require('../lib');
+
+const {
+    validateJwtWithPermissions, checkPermissionRights, checkPermissionRightsStrict
+} = jwtPermissions;
+
+
+const { privateKey } = JWTMockServer;
+
+const mockPayload = { merchantId: '123456789abc', permissions: '00F' };
+let expectedPayload = {
+    merchantId: mockPayload.merchantId,
+    decodedPermissions:
+        [false, false, false, false, false, false, false, false, true, true, true, true]
+};
+
+let signedToken;
+const port = 3456;
+let server;
+let app;
+let router;
+
+
+let issuer = new JWTIssuer({ issuer: 'Storyous s.r.o.', privateKey });
+const jwtTokenSign = (payload) => issuer.createToken(payload, []);
+
+const startServerWithToken = ({ publicKeyUrl }) => {
+    app = new Koa();
+    app.use(koaBody({ includeUnparsed: true }));
+    router = routerFactory();
+    router.get('/', validateJwtWithPermissions({ publicKeyUrl }),
+        (ctx) => {
+            ctx.body = { jwtPayload: ctx.state.jwtPayload, permissions: ctx.state.permissions };
+            ctx.status = 200;
+        }
+    );
+    app.use(router.routes());
+    server = app.listen(port);
+};
+
+
+describe('JWT authorization', () => {
+    beforeEach(() => startServerWithToken({}));
+    afterEach(async () => server.close());
+    it('should decode token', async () => {
+        signedToken = jwtTokenSign(mockPayload, privateKey);
+        const response = await fetch.json(`http://127.0.0.1:${port}/`,
+            {
+                method: 'get',
+                headers: {
+                    'Content-Type': 'application/json',
+                    authorization: `"Bearer" ${signedToken}`
+                }
+            });
+        assert.deepStrictEqual(response.jwtPayload.decodedPayload, expectedPayload);
+
+
+    });
+    it('should fail decode token with invalid publicTesting key', async () => {
+        signedToken = jwtTokenSign(mockPayload, privateKey);
+        await server.close();
+        startServerWithToken({ publicKeyUrl: 'http://127.0.0.1:3010/getInvalidPublicKey' });
+        let error;
+        try {
+            await fetch.json(`http://127.0.0.1:${port}/`,
+                {
+                    method: 'get',
+                    headers: {
+                        'Content-Type': 'application/json',
+                        authorization: `"Bearer" ${signedToken}`
+                    }
+                });
+        } catch (e) {
+            error = e;
+        }
+        assert(error instanceof Error);
```

Comment:
This is not enough, we should check it returns the error we expect not just some random one.

Reply:
+ Do the same on other tests where is just `assert(error instanceof Error);`

Reply:
done